Purchases/Caching/RCDeviceCache.m

```diff
@@ -62,27 +66,38 @@ - (void)clearCachesForAppUserID:(NSString *)appUserID
 
 - (BOOL)isCacheStale {
     NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
-    return !(self.cachesLastUpdated != nil && timeSinceLastCheck < 60. * 5);
+    return !(self.cachesLastUpdated != nil && timeSinceLastCheck < CACHE_DURATION_IN_SECONDS);
+}
+
+- (BOOL)isOfferingsCacheStale {
+    return self.offeringsCachedObject.isCacheStale;
+}
+
+- (RCOfferings * _Nullable)cachedOfferings {
+    return self.offeringsCachedObject.cachedInstance;
 }
 
 - (void)resetCachesTimestamp
 {
-    self.cachesLastUpdated = [NSDate date];
+    NSDate *now = [NSDate date];
+    self.cachesLastUpdated = now;
+    [self.offeringsCachedObject updateCacheTimestampWithDate:now];
 }
 
 - (void)clearCachesTimestamp
 {
     self.cachesLastUpdated = nil;
+    [self.offeringsCachedObject clearCacheTimestamp];
 }
 
 - (void)cacheOfferings:(RCOfferings *)offerings
 {
-    self.cachedOfferings = offerings;
+    [self.offeringsCachedObject cacheInstance:offerings date:[NSDate date];
```

Comment:
setting the date here is a bit of a change in behavior, so I'd like some feedback on this: 

there are currently two instances of setting cached offerings:
- you want to get something from cache, like purchaser info, and the cache is stale. The cache timestamp is updated to right now, and then we fetch stuff from the backend. If something goes wrong, cache is set to nil so that it'll be considered stale again. all good. 
- you want to get offerings with `updateOfferingsCache:completion`, and there are no cached offerings. 
looking at the code, it seems that in this case, the cache date isn't updated (which kinda makes sense since the other stuff in cache isn't updated), but the cached offerings are set. 


This won't be a huge problem, but it seems like a bug, right? 
in the case where: 
1. cache is stale
2. you get offerings, they get cached but date isn't updated
3. you get purchaser info
on step 3, you'll re-fetch because the date hasn't been set, so you fetch everything, including offerings a second time. 
also, if before doing step 3, offerings cache is technically stale, while offerings are correctly cached. 
is this a correct interpretation? 
if it is, it's solved by updating the date here, since we now have a separate date for offerings and the rest, so two birds with one stone.